src/routes/mods.rs

```diff
@@ -130,17 +132,32 @@ pub async fn mod_delete(
     pool: web::Data<PgPool>,
     config: web::Data<SearchConfig>,
 ) -> Result<HttpResponse, ApiError> {
-    check_is_moderator_from_headers(
-        req.headers(),
-        &mut *pool
-            .acquire()
+    let user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let id = info.into_inner().0;
+
+    if user.role != Role::Moderator || user.role != Role::Admin {
+        let mod_item = database::models::Mod::get(id.into(), &**pool)
             .await
-            .map_err(|e| ApiError::DatabaseError(e.into()))?,
-    )
-    .await
-    .map_err(|_| ApiError::AuthenticationError)?;
+            .map_err(|e| ApiError::DatabaseError(e.into()))?
+            .ok_or_else(|| ApiError::InvalidInputError("Invalid Mod ID specified!".to_string()))?;
+        let team_member = database::models::TeamMember::get_from_user_id(
+            mod_item.team_id,
+            user.id.into(),
+            &**pool,
+        )
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?
+        .ok_or_else(|| ApiError::InvalidInputError("Invalid Mod ID specified!".to_string()))?;
+
+        if Permissions::from_bits_truncate(team_member.permissions as u64) & Permissions::DELETE_MOD
+            != Permissions::DELETE_MOD
```

Comment:
This should use `from_bits` rather than `from_bits_truncate` and handle the error.  This should also use `.contains(Permissions::DELETE_MOD)`, as it is more concise.

Reply:
I switched to contains. Why from bits? There is no need for an error, as we can still get the permissions out of the bits. Some languages may use the on bit as default, so we shouldn't throw an error for that

Reply:
This should be `from_bits` because it should fail if there is anything incorrect with the permissions bitset.  For example, if we add new permissions and accidentally run on an older version of the backend with newer permissions in the database, we want to be careful to handle that well.  In this case it should be fine to use `from_bits_truncate` as it's only checking permissions, but everywhere else should use `from_bits` and check for errors.  If we don't check for errors, we may lose data, as the bitset will silently truncate the newer permissions.

Reply:
Isn't this line checking for permissions though, for the editing and create I can change to from bits.